tests/functional/deprecations/model_deprecations.py

```diff
@@ -0,0 +1,78 @@
+import pytest
+
+from dbt.cli.main import dbtRunner
+
+deprecated_model__yml = """
+version: 2
+
+models:
+  - name: my_model
+    description: deprecated
+    deprecation_date: 1999-01-01
+"""
+
+deprecating_model__yml = """
+version: 2
+
+models:
+  - name: my_model
+    description: deprecating in the future
+    deprecation_date: 2999-01-01
+"""
```

Comment:
Are `deprecation_date` formats like the following accepted?
- `2022-04-08T22:00:00.0-04:00`
- `2022-04-08 22:00:00.0-04:00`
- `2022-04-08T22:00:00.0`
- `2022-04-08 22:00:00.0`

If so, could we add test cases for the relevant ones?

@MichelleArk Could you or someone else add which cases do or don't work in the docs issue [here](https://github.com/dbt-labs/docs.getdbt.com/issues/3357)?

Reply:
@dbeatty10 We are using PyYAML to parse JSON, and relying on its date detection and parsing for now, which claims to follow the YAML spec. I'll figure out what we can say definitively about the dates we accept.